### DISCUSSION TITLE
Rules that only differ by their separator modifier does not parse

### DISCUSSION THREAD

**Discussion by hallgrim** [textX/textX#328] [Q&A]
I have a simple model:
```
seq1: these & are & good & friends
seq2: those | exclusive | ones
```
I try to parse it with this meta model:
```
Document: sequences*=Sequence;
Sequence: SequenceAnd | SequencePipe;
SequenceAnd: ID ':' values+=ID[eolterm '&'];
SequencePipe: ID ':' values+=ID[eolterm '|'];
```
The meta model checks OK, but when I try to use it on the model above I get:
```
Expected '&' or ID or ID or EOF at position .\model.txt:(2, 13) => 'q2: those *| exclusiv'.
```

To reproduce:
[model.txt](https://github.com/igordejanovic/textX/files/1461309/model.txt)
and
[meta.tx.txt](https://github.com/igordejanovic/textX/files/1461315/meta.tx.txt)
and run
```textx check meta.tx.txt model.txt```

As far as I can understand from reading the grammar this should work, but it might be a mistake on my side.

**Comment by hallgrim** [comment]
I worked around this issue with an explicit separator in the rule:
From: 
```
SequenceAnd: ID ':' values+=ID[eolterm '&'];
```
To: 
```
SequenceAnd: ID ':' values=ID '&' values+=ID[eolterm '&'];
```
This will capture the first value in the sequence, then an explicit separator and then the rest of the sequence.

**Comment by igordejanovic** [comment]
This is indeed an interesting example.
If you run the example in debug mode with:
```
textx -d check meta.tx.txt model.txt
```
you will notice that the second line, actually just a part `seq2: those` is parsed with `SequenceAnd` as it is perfectly valid from the standpoint of that rule. You can have only one element of `values`. Thus the rule succeeds and the parser expect next to see either a new Sequence or end of file which doesn't happen.
This behaviour of textX is perfectly valid.

Your workaround is fine. In cases like these you might also use `not` or `and` predicate. For example:
```
Document: sequences*=Sequence;
Sequence: SequenceAnd | SequencePipe;
SequenceAnd: ID ':' &(ID '&') values+=ID[eolterm '&'];
SequencePipe: ID ':' values+=ID[eolterm '|'];
```
Notice the `&(ID '&')` part. This is `and` predicate which must match at that place in order to proceed with this rule but will not consume the input. If there is no `&` after the ID `SequenceAnd` will fail and `SequencePipe` will be tried and it will succeed even if there is only one element of a Sequence thus `SequencePipe` will be a rule matched if there is only one element of the sequence, just like in your solution.
